### ISSUE TITLE
TypeHintDeclarationSniff does not detect Closure

### ISSUE THREAD

**Issue by carusogabriel** [slevomat/coding-standard#230]
_This is a question, that could become a bug report_ :smile:

I have this piece of `PHP` code:
```php
private function testing(): void
{
    $array = [1, 2, 3];

    $filteredArray = array_filter($array, function ($current): bool {
        return $current > 1;
    });

    die(var_dump($filteredArray));
}
```

And I have the `TypeHintDeclarationSniff` sniff loaded. If I remove the return type-hint `void` from `testing` function, the `TypeHintDeclarationSniff.MissingReturnTypeHint` detects it. On the other hand, if I remove `bool` from the `Closure`, it does not detect. 

Am I using the correct `Sniff`, or no, this is a bug?

I find it in [`rectorphp/rector#233`](https://github.com/rectorphp/rector/pull/233) :wrench:

**Comment by Majkl578** [comment]
I believe this can't be reliably fixed/enforced by PHPCS alone. Since PHPCS is not a static analysis tool, it has no knowledge about types so it can't reliably make any assumptions about typing and would lead to false positives. Also it can't be usual _native_/_documented_ decision as for method parameters, since closures have no documentation standard. The only reliable check is for `void` - that is when no return or yield exists inside the closure.

**Comment by kukulich** [comment]
Thanks @Majkl578 for perfect explanation.

**Comment by carusogabriel** [comment]
Thanks @Majkl578 👏
I’ll keep track of it in another way!
